packages/ariakit/src/tab/tab-panel.ts

```diff
@@ -45,13 +45,27 @@ export const useTabPanel = createHook<TabPanelOptions>(
 
     const [hasTabbableChildren, setHasTabbableChildren] = useState(false);
 
-    useEffect(() => {
+    const watchForTabbableChildren = useCallback(() => {
       const element = ref.current;
       if (!element) return;
       const tabbable = getAllTabbableIn(element);
       setHasTabbableChildren(!!tabbable.length);
     }, []);
 
+    useEffect(() => {
+      watchForTabbableChildren();
+
+      const observer = new MutationObserver(watchForTabbableChildren);
+      const element = ref.current;
+      if (!element) return;
+      observer.observe(element, {
+        attributes: true,
+        childList: true,
+        subtree: true,
+      });
+      return () => observer.disconnect();
+    }, [watchForTabbableChildren]);
```

Comment:
The watchForTabbableChildren function could be inside the effect so we can avoid this extra useCallback.